### PR TITLE
66 update endpoints based on role

### PR DESCRIPTION
**Pull request by ligopsychakis-michalis**
@zannis we created a new function at the bottom of routes.js to check if the login user is admin or not
and we used it to routes that ticket wanted us to update...

*we found problem with passwords.. we saw that keyword "delete" doesn't actually delete them... we didn't found a way to overcome this problem.. can you give us an advice for that??

### CODE REVIEW

**Comment by student0b4dc0d3** [comment]
You could always "EXCLUDE" any non-desired fields from being retrieved, especially if the result dataset includes more than 1 document, thus you would **not need** to iterate the complete dataset to remove unwanted fields after.

Also, always keep in mind that the resulting dataset is always an array of documents, even when you ask for 1 document.

I hope this helps a little.
Have a great day.

**Comment by zannis** [comment]
As Manolis mentioned, you can use the mongoose select chained function to omit or get any specific fields like the password.
https://mongoosejs.com/docs/api.html#query_Query-select

File: backend/src/routes.js [hunk 1]
```diff
@@ -298,12 +297,12 @@ function createJWTs(id, role) {
 }
 
 // function to check if user is Admin or not
-function checkAdmin(auth) {
+function isAdmin(auth) {
     const accessToken = auth.split(' ')[1];
     const payload = jwt.decode(accessToken);
```
Comment (zannis) [inline backend/src/routes.js:302]: You don't need to decode the access token again. This has already been done by the express-jwt middleware.
This middleware has exposed `req.user` which contains the payload. So you can check the role directly there.

Comment (ligopsychakis-michalis) [reply backend/src/routes.js:302]: a ok.. I didn't knew it..
so it's no need to use the function isAdmin too, right??

**Pull request merged**